src/MCPServer/lib/workflow.py

```diff
@@ -0,0 +1,315 @@
+# -*- coding: utf-8 -*-
+
+"""Workflow decoder and validator.
+
+The main function to start working with this module is ``load``. It decodes the
+JSON-encoded bytes and validates the document against the schema.
+
+    >>> import workflow
+    >>> with open("workflow.json") as fp:
+            wf = workflow.load(fp)
+
+If the document cannot be validated, ``jsonschema.ValidationError`` is raised.
+Otherwise, ``load`` will return an instance of ``Workflow`` which is used in
+MCPServer to read workflow links that can be instances of three different
+classes ``Chain``, ``Link`` and ``WatchedDir``. They have different method
+sets.
+"""
+
+from __future__ import unicode_literals
+
+import importlib
+import json
+import os
+import pprint
+
+from django.utils.six import text_type, python_2_unicode_compatible
+from jsonschema import validate, FormatChecker
+
+from main.models import Job
+
+
+_LATEST_SCHEMA = "workflow-schema-v1.json"
+
+_UNKNOWN_TRANSLATION_LABEL = "<unknown>"
+
+_DEFAULT_LANG = "en"
```

Comment:
> An aside, having worked through this a bit more _DEFAULT_LANG could be changed to _FALLBACK_LANG. I feel this would be more semantically accurate for the reader. But that might be a moot point if we change this code a bit more.

Changed.

> I don't want to go too wild with suggestions how to resolve this here as you probably have a clearer idea of what could be done, but happy to chat more. I think it could be resolved in workflow.json with code changes too, but I could imagine other ways in code alone.

It's hard to solve this problem in the Dashboard because it doesn't have access to the workflow and it shouldn't. I'll think about this more but I'll probably be adding a new RPC method that returns the list given the unit UUID.

Reply:
See `6) Update Dashboard - microservice list`, I think it solves the problem nicely by pushing the responsibility to MCPServer. I think that `getUnitStatus` could potentially be used in other places too.